fenix/app/src/main/java/org/mozilla/fenix/shopping/ReviewQualityCheckFeature.kt

```diff
@@ -4,28 +4,49 @@
 
 package org.mozilla.fenix.shopping
 
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapNotNull
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.flowScoped
 import mozilla.components.support.base.feature.LifecycleAwareFeature
-import org.mozilla.fenix.nimbus.FxNimbus
 
 /**
  * Feature implementation that provides review quality check information for supported product
  * pages.
  *
- * @param onAvailabilityChange Invoked when availability of this feature changes based on feature
+ * @property browserStore Reference to the application's [BrowserStore].
+ * @property shoppingExperienceFeature Reference to the [ShoppingExperienceFeature].
+ * @property onAvailabilityChange Invoked when availability of this feature changes based on feature
  * flag and when the loaded page is a supported product page.
  */
 class ReviewQualityCheckFeature(
+    private val browserStore: BrowserStore,
+    private val shoppingExperienceFeature: ShoppingExperienceFeature = ShoppingExperienceFeature(),
     private val onAvailabilityChange: (isAvailable: Boolean) -> Unit,
 ) : LifecycleAwareFeature {
+    private var scope: CoroutineScope? = null
 
     override fun start() {
-        val isFeatureEnabled = FxNimbus.features.shoppingExperience.value().enabled
-        // Update to use product page detector api in Bug 1840580
-        val isSupportedProductPage = false
-        onAvailabilityChange(isFeatureEnabled && isSupportedProductPage)
+        if (shoppingExperienceFeature.isEnabled().not()) {
+            onAvailabilityChange(false)
+            return
+        }
+
+        scope = browserStore.flowScoped { flow ->
+            flow.mapNotNull { it.selectedTab }
+                .map { it.isProductUrl }
+                .distinctUntilChanged()
+                .collect {
+                    onAvailabilityChange(it)
+                }
```

Comment:
drive by shower thought: curious if this works
```suggestion
                .collect(onAvailabilityChange)
```

Reply:
Nice! 👍